main.rb

```diff
@@ -0,0 +1,94 @@
+module Enumerable
+  def my_each
+    index = 0
+
+    while index < size
+      yield self[index]
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    item = 0
+    index = 0
+
+    while index < size
+      yield self[item], index
+      index += 1
+      item += 1
+    end
+  end
+
+  def my_select
+    arr = []
+
+    my_each do |item|
+      arr.push(item) if yield item
+    end
+    arr
+  end
+
+  def my_all?
+    bool = true
+
+    my_each do |item|
+      bool = false unless yield item
+    end
+    bool
+  end
+
+  def my_any?
+    bool = false
+
+    my_each do |item|
+      bool = true if yield item
+    end
+    bool
+  end
+
+  def my_none?
+    bool = true
+
+    my_each do |item|
+      bool = false if yield item
+    end
+    bool
+  end
+
+  def my_count
+    count = 0
+
+    my_each do |item|
+      count += 1 if yield item
+    end
+    count
+  end
+
+  def my_map
+    arr = []
+
+    my_each do |item|
+      arr.push(yield item)
+    end
+    arr
+  end
+
+  def my_inject(memo = 0)
+    arr = to_a
+
+    if memo.zero?
+      memo = arr[0]
+      arr.shift
+    end
+
+    my_each do |item|
+      memo = yield memo, item
+    end
+    memo
+  end
+end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- when you pass a block to perform an operation, your method is not given the right result. Check the value of memo because it's affecting the operation.

- Your method should work with symbols as arguments (:+) and perform the right operation.